src/macros.rs

```diff
@@ -0,0 +1,31 @@
+//Based on smoltcp's macro_rules
+#[cfg(not(test))]
+#[cfg(feature = "log")]
+#[macro_export]
+macro_rules! net_log {
```

Comment:
I don't think `net_log` is a good name in our context. Perhaps just `log` or maybe better yet - `svc_log`.